clients/dataset/dataset.go

```diff
@@ -83,60 +43,42 @@ var _ error = ErrInvalidDatasetAPIResponse{}
 
 // Client is a dataset api client which can be used to make requests to the server
 type Client struct {
-	cli           RHTTPClient
-	ctxcli        RCHTTPClient
-	url           string
-	internalToken string
+	cli common.RCHTTPClienter
+	url string
 }
 
-// New creates a new instance of Client with a given dataset api url
-func New(datasetAPIURL string) *Client {
+// NewAPIClient creates a new instance of Client with a given dataset api url and the relevant tokens
+func NewAPIClient(datasetAPIURL, serviceToken, xDownloadServiceToken, florenceToken string) *Client {
```

Comment:
It seems odd to have a Florence token injected into the constructor - won't this potentially change for each request if there is a different user? I would imply that a new client is needed for each request?

Reply:
you're right, I think I got carried away with fields to put into the client, I'll take it out